NodePool API: rename spec.tunedConfig to spec.tuningConfig

dagrayvid opened the pull request:
**What this PR does / why we need it**:
This PR renames the spec.TunedConfig field of the NodePool API to spec.tuningConfig. This is needed to keep the name general because we will likely use this field to reference PerformanceProfiles later on when we enable the PAO / PerformanceProfile functionality (~4.13 timeframe). 

As discussed in the HyperShift project sync meeting, we would like to make the change while the NodePool API is still in alpha.

/cc @jlojosnegros

**Which issue(s) this PR fixes**:
Related to [PSAP-742](https://issues.redhat.com/browse/PSAP-742) and [CNF-6291](https://issues.redhat.com/browse/CNF-6291)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [x] This change includes docs. 
- [x] This change includes unit tests.

dagrayvid commented:
/assign @enxebre 

/hold
for a little more manual testing...

dagrayvid commented:
/retest

enxebre commented on hypershift-operator/controllers/nodepool/nodepool_controller.go:
```diff
@@ -81,8 +81,8 @@ const (
 
-	tunedConfigKey                 = "tuned"
-	tunedConfigMapLabel            = "hypershift.openshift.io/tuned-config"
+	tuningConfigKey                = "tuned"
```
does tuned as key still makes sense?

enxebre commented:
/approve
/lgtm
can you show a grep 'tunedConfig' canceling the hold?

jlojosnegros commented:
/retest

jlojosnegros commented:
/lgtm

jlojosnegros commented:
/hold 
until @enxebre has an opportunity to review, 
and maybe till openshift/cluster-node-tuning-operator#491 makes it to the ci release?

dagrayvid commented:
/retest

dagrayvid commented:
@enxebre, here is the grep:
```
$ grep -r -i -n --exclude-dir .git --exclude-dir vendor tunedconfig .
./hypershift-operator/controllers/nodepool/nodepool_controller_test.go:854:      tunedConfig:
./hypershift-operator/controllers/nodepool/nodepool_controller_test.go:902:      tunedConfig:
```

The two occurrences of tunedConfig in nodepool_controller_test.go are from the spec.recommend.operand.tunedConfig field of the Tuned CRs. This is a real field in the Tuned spec which controls some configuration of the TuneD daemon. After this PR, we won't be overloading the term anymore. :)

dagrayvid commented:
/retest

dagrayvid commented:
/retest

dagrayvid commented:
/retest

dagrayvid commented:
Updated the PR because the `tuningConfigKey` in test/e2e/nodepool_nto_machineconfig_test.go was not updated from tuned to tuning. Also added the PR description to the commit message.

dagrayvid commented:
@enxebre @jlojosnegros, this should now be ready to merge.

jmencak commented:
Thank you for the changes, David.
/lgtm

dagrayvid commented:
/hold 

@enxebre Please unhold if you think this PR is good to merge!

jmencak commented:
/retest

enxebre commented:
/hold cancel
